airflow-core/src/airflow/models/connection.py

```diff
@@ -348,7 +348,14 @@ def get_password(self) -> str | None:
                     f"Can't decrypt encrypted password for login={self.login}  "
                     f"FERNET_KEY configuration is missing"
                 )
-            return fernet.decrypt(bytes(self._password, "utf-8")).decode()
+            try:
+                return fernet.decrypt(bytes(self._password, "utf-8")).decode()
+            except Exception as e:
+                from cryptography.fernet import InvalidToken
+
+                if isinstance(e, InvalidToken):
+                    return self._password
+                raise
```

Comment:
I think it can be simplified to
```suggestion
            except InvalidToken as e:
                 return self._password
```
...then it will re-raise automatically if the exception is of a different type.

Reply:
Addressed this